core/modules/utils/dom/popup.js

```diff
@@ -163,7 +170,9 @@ Popup.prototype.cancel = function(level) {
 				popup.wiki.deleteTiddler(popup.title);
 			} else {
 				popup.wiki.deleteTiddler($tw.utils.parseTextReference(popup.title).title);
-        		}
+			}
+			var index = this.findPopup(this.registeredPopups,popup.title);
```

Comment:
I do have a bad feeline about the `index` variable. If `.findPopup()` returns `-1` it will remove the last element of `registeredPopups`. This may be intended, but from my point of view that's a "side-effect" and seems to be error prone and complex to maintain. ... BUT I'm not 100% sure what's intended.